src/modelframe.jl

```diff
@@ -172,19 +172,40 @@ function setcontrasts!(mf::ModelFrame, new_contrasts::Dict)
 end
 setcontrasts!(mf::ModelFrame; kwargs...) = setcontrasts!(mf, Dict(kwargs))
 
+convertinsupport(v::AbstractVector, d::Distribution) = Vector{partype(d)}(v)
+
+function convertinsupport(v::Union{AbstractCategoricalVector, AbstractNullableCategoricalVector},
+    d::Bernoulli)
+    vdrp = droplevels!(copy(v))
+    levs = CategoricalArrays.levels(vdrp)
+    llv = length(levs)
+    if llv < 2
+        zeros(vdrp, partype(d))
+    elseif llv == 2
+        partype(d)[get(x) == levs[2] for x in vdrp]
+    else
+        throw(ArgumentError("length(levels(droplevels!(v))) = $llv, should be ≤ 2"))
```

Comment:
Maybe the message should rather mention the Bernouilli distribution and the number of levels present in the response variable? If people fitting a GLM see this without calling `convertinsupport` themselves, they don't know what is `v`.